### PR TITLE
`azurerm_mssql_database` - add support for `long_term_retention_policy` and `short_term_retention_policy`

### LINKED ISSUE #1802: Resources and Datasources for SQL Server Backups
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We would like to use terraform to manage our sql server backups: https://docs.microsoft.com/en-us/rest/api/sql/backupshorttermretentionpolicies/createorupdate

### New or Affected Resource(s)

Should be resources and datasources for Backup Policies

### References

https://docs.microsoft.com/en-us/rest/api/sql/backupshorttermretentionpolicies/createorupdate

### PR DESCRIPTION
**Pull request by jackofallops**
encompasses and supersedes #6194 

closes #1802

### CODE REVIEW

File: azurerm/internal/services/mssql/mssql_database_resource.go [hunk 1]
```diff
@@ -485,6 +538,7 @@ func resourceArmMsSqlDatabaseRead(d *schema.ResourceData, meta interface{}) erro
 		} else if props.ReadScale == sql.DatabaseReadScaleDisabled {
 			d.Set("read_scale", false)
 		}
+		skuName = *props.CurrentServiceObjectiveName
```
Comment (mbfrahry) [inline azurerm/internal/services/mssql/mssql_database_resource.go:541]: Should we check for nil here?

**Review by mbfrahry — CHANGES_REQUESTED** [review]
This looks good but I did have one point to chat about around defaulting values

File: website/azurerm.erb [hunk 2]
```diff
@@ -1411,7 +1411,7 @@
                 </li>
 
                 <li>
-                  <a href="/docs/providers/azurerm/r/mssql_database_extended_auditing_policy.html">azurerm_mssql_database_extended_auditing_policy</a>
+                  <a href="/docs/providers/azurerm/r/mssql_database.html">azurerm_mssql_database</a>
```
[the remark this answers is not in the document]
Comment (jackofallops) [reply website/azurerm.erb:1414]: Thanks!  Look like I skipped a commit by accident when rebasing for conflicts.

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Comment by jackofallops** [comment]
Tests passing (failures unrelated to change / transient)
image

**Pull request merged**